oeps/oep-0028-content-theming-in-react.rst

```diff
@@ -0,0 +1,334 @@
+
+1. Inheritance becomes necessary when you want to customize a component, but there is no support via props. An example of this can be ``SiteLogoCircle`` component, where the user wants to a Trademark logo next to the image. Since ``SiteLogoCircle`` does not contain support for trademark logo, it needs to extend via inheritance as follows
+
+.. code-block:: js
+
+    class SiteLogoCircle extends React.PureComponent {
+        render() {
+            <CircleLogo>
+                <Image src={this.props.image} />
+            </CircleLogo>
+        }
+    }
+
+    // if props were present
+    class SiteLogoCircle extends React.PureComponent {
+        render() {
+            <CircleLogo>
+                <Image src={this.props.image} />
+                {this.props.trademark ? <TradeMarkLogo />:  null}
+            </CircleLogo>
+        }
+    }
+
+    // via inheritance
+    class SiteLogoCircleWithTitle extends SiteLogo {
+        render() {
+            <CircleLogo>
+                <Image src={this.props.image} />
+                <TradeMarkLogo />
+            </CircleLogo>
+        }
+    }
+
+2. Inheritance becomes necessary when the original author is not aware of the customizations a component can have. An example of this can be Card component, where the author has a CardColor component which gives a background color to the upper half portion of the Card. Now in the custom theme, we want to show an image instead of the background color. This can be done via inheritance
+
+.. code-block:: js
+
+    class Card extends React.PureComponent {
+
+        viewCardDetail(courseId) {
+            // open course detail page
+        }
+
+        render() {
+            <Card>
+                <CardColor color={this.props.color} />
+                <CardBody>
+                    <CardTitle />
+                    <CardText />
+                    <Button onClick={this.viewCardDetail(this.props.courseId) />}
+                </CardBody>
+            </Card>
+        }
+    }
+
+    // overridden version
+    class ImageCard extends Card {
+        render() {
+            <Card>
+                <CardImage image={this.props.imageUrl} />
+                <CardBody>
+                    <CardTitle />
+                    <CardText />
+                    <Button onClick={this.viewCardDetail(this.props.courseId) />}
+                </CardBody>
+            </Card>
+        }
+    }
+
+3. Inheritance often becomes necessary and useful, when we want to override the rendering functionality of any component, and still maintaining access to lifecycle code. Overriding functionality can include removing, re-ordering, replacing or inserting children. An example of this can be Navbar, where the default Navbar has a SearchForm which is left aligned. This Navbar component can now be inherited to place SearchForm as right aligned.
```

Comment:
> Overriding functionality can include removing, re-ordering, replacing or inserting children.

I think this covers all three examples? Example 1 inserts a child, example 2 replaces one child, and example 3 reorders it. All three examples are basically only rewriting the `render` function, so I don't think we need all three of them - example 3 is enough. Another good example of inheritance is the one from before this section, where we are overriding `extraNavLinks`. I think we should put these two examples in the same section, and remove redundant examples 1 & 2 from this section.

Reply:
Sure, I have refactored the point 3 and removed point 1 and 2. Thanks :)